src/internal/connector/exchange/api/contacts.go

```diff
@@ -125,32 +130,42 @@ func (c Contacts) EnumerateContainers(
 		ChildFolders()
 
 	for {
-		resp, err = builder.Get(ctx, ofcf)
+		resp, err := builder.Get(ctx, ofcf)
 		if err != nil {
-			return errors.Wrap(err, support.ConnectorStackErrorTrace(err))
+			return clues.Stack(err).WithClues(ctx).WithAll(graph.ErrData(err)...)
 		}
 
 		for _, fold := range resp.GetValue() {
+			if errs.Err() != nil {
+				return errs.Err()
+			}
+
 			if err := checkIDAndName(fold); err != nil {
-				errs = multierror.Append(err, errs)
+				errs.Add(clues.Stack(err).WithClues(ctx).WithAll(graph.ErrData(err)...))
 				continue
 			}
 
+			fctx := clues.AddAll(
+				ctx,
+				"container_id", ptr.Val(fold.GetId()),
+				"container_display_name", ptr.Val(fold.GetDisplayName()))
+
 			temp := graph.NewCacheFolder(fold, nil, nil)
 			if err := fn(temp); err != nil {
-				errs = multierror.Append(err, errs)
+				errs.Add(clues.Stack(err).WithClues(fctx).WithAll(graph.ErrData(err)...))
 				continue
 			}
 		}
 
-		if resp.GetOdataNextLink() == nil {
+		link, ok := ptr.ValOK(resp.GetOdataNextLink())
+		if !ok {
 			break
 		}
 
-		builder = users.NewItemContactFoldersItemChildFoldersRequestBuilder(*resp.GetOdataNextLink(), service.Adapter())
+		builder = users.NewItemContactFoldersItemChildFoldersRequestBuilder(link, service.Adapter())
 	}
 
-	return errs.ErrorOrNil()
+	return errs.Err()
```

Comment:
Do we no longer want to use the `OrNil` package here?

Reply:
Nope.  `ErrorOrNil` is a silly hack and shouldn't be needed.